src/sentry/models/organization.py

```diff
@@ -188,12 +191,20 @@ def __str__(self):
 
     def save(self, *args, **kwargs):
         if not self.id:
-            self.id = snowflake_id_generation()
+            self.id = snowflake_id_generation(self.snowflake_redis_key)
         if not self.slug:
             lock = locks.get("slug:organization", duration=5)
             with TimedRetryPolicy(10)(lock.acquire):
                 slugify_instance(self, self.name, reserved=RESERVED_ORGANIZATION_SLUGS)
-        super().save(*args, **kwargs)
+        try:
+            with transaction.atomic():
+                super().save(*args, **kwargs)
+        except IntegrityError:
+            if self.snowflake_retry_counter == settings.MAX_REDIS_SNOWFLAKE_RETY_COUNTER:
```

Comment:
@maxiuyuan this might be a better if guard. Since we don't want to raise an exception on the fifth attempt.

```suggestion
            if self.snowflake_retry_counter > settings.MAX_REDIS_SNOWFLAKE_RETY_COUNTER:
```